Added getters for JsonPatch and Operations

GavinF17 opened the pull request:
Hoping this gets picked up & released due to recent activity by @Capstan :)

- Added JsonPatch.getOperations()
- Added field getters for operations
  - DualPathOperation.getFrom()
  - JsonPatchOperation.getOp() and JsonPatchOperation.getPath()
  - PathValueOperation.getValue()

Personal use case is needing to determine which class to patch when using subtypes, and the type info being held in a field which may be modified by a replace operation.

Solves #33 
Supercedes #45

Spellmaker commented:
I'm also highly interested in this PR getting merged, getters would be really useful for examining the operations (in my use-case: checking for certain paths to discourage changes to some part of the json document)

serac commented:
My use case is similar to @Spellmaker but notably different. We have a JSON model that maps to database models where only some fields are mutable via our REST API. We need to be able to example a patch request in order to validate that the JSON paths are mutable according to business policy. While we could do this with raw JSON and then parse it into a `JsonPatch` object after validation, it would be far more convenient to work with the parsed version where the operation and path are easily accessible on a `JsonPatch` object.

As far as this PR, it seems innocuous as `operations` is final and wrapped in an immutable collection, so there's no risk of clients mutating the internal state by exposing it via a getter. I hope you'll consider this enhancement.

Capstan commented on src/main/java/com/github/fge/jsonpatch/DualPathOperation.java:
```diff
@@ -72,6 +72,10 @@ public final void serializeWithType(final JsonGenerator jgen,
         serialize(jgen, provider);
     }
 
+    public JsonPointer getFrom() {
```
Why not `final`? Here and elsewhere.

GavinF17 replied:
Should have had this in, implemented now

Capstan commented on src/main/java/com/github/fge/jsonpatch/JsonPatchOperation.java:
```diff
@@ -94,6 +94,14 @@ protected JsonPatchOperation(final String op, final JsonPointer path)
         throws JsonPatchException;
 
+    public String getOp() {
+        return op;
```
Should this protectively copy the string?

GavinF17 replied:
I wouldn't have thought this is necessary as `String` is immutable, if I'm not thinking of something can you please explain?

Capstan replied:
Ah, you're right.

Capstan commented on src/main/java/com/github/fge/jsonpatch/PathValueOperation.java:
```diff
@@ -73,6 +73,10 @@ public final void serializeWithType(final JsonGenerator jgen,
     }
 
+    public JsonNode getValue() {
+        return value;
```
This is the one where if `JsonNode` is mutable, then callers can change the value out from under the `Operation`. Should we not protectively `deepCopy` here?

GavinF17 replied:
Added `deepCopy` as suggested, I didn't realise `JsonNode` is mutable so good spot!

GavinF17 commented:
> Looks good overall.

Thanks for the feedback, addressed all comments with [one outstanding](https://github.com/java-json-tools/json-patch/pull/79#discussion_r464405619)

elalex2021 commented:
I am using v.1.13 (https://mvnrepository.com/artifact/com.github.java-json-tools/json-patch/1.13), but not able to get the patch operations using method getOperations(). You need to add it to public interface Patch. Could you please do that ASAP? Thanks.

elalex2021 commented:
I am using v.13 (https://mvnrepository.com/artifact/com.github.java-json-tools/json-patch/1.13), but still not able to use method getOperations(). 

![image](https://user-images.githubusercontent.com/84472794/127044967-b7b45e3d-efb1-464d-8b80-758da35c9d2e.png)

ceenlrnjim commented:
We are also interested in having the getOperations accessor and see that it isn't in 1.13 - are there plans for another release that will pick up this change?
